src/models/fragment.js

```diff
@@ -0,0 +1,50 @@
+import { FragmentStatus, RecordType } from '../constants/enums';
+import { validate } from 'uuid';
+import { getUKTimestamp } from '../services/time';
+import { addChangesToUpdateParams } from '../utilities/dynamodb-helper';
+
+const fieldsAllowedToUpdate = [
+  'TransferStatus',
+  'OutboundMessageId',
+  'AcknowledgementReceivedAt',
+  'AcknowledgementTypeCode',
+  'AcknowledgementDetail'
+];
+
+export const buildFragmentUpdateParams = (inboundConversationId, inboundMessageId, changes) => {
+  validateIds(inboundConversationId, inboundMessageId);
+
+  const params = {
+    Key: {
+      InboundConversationId: inboundConversationId,
+      Layer: [RecordType.FRAGMENT, inboundMessageId].join('#')
+    },
+    UpdateExpression: `set UpdatedAt = :now`,
+    ExpressionAttributeValues: {
+      ':now': getUKTimestamp()
+    }
+  };
+
+  return addChangesToUpdateParams(params, changes, fieldsAllowedToUpdate);
+};
+
+const validateIds = (conversationId, messageId) => {
+  const uuidsAreValid = validate(conversationId) && validate(messageId);
+  if (!uuidsAreValid) {
+    throw new Error('received invalid uuid as either conversationId or messageId');
```

Comment:
Apologies if I'm incorrect, been a while since I've used JavaScript 😂!

Should this be:

```suggestion
    throw new ValidationError('received invalid uuid as either conversationId or messageId');
```

As per your `ValidationError` declaration in `errors.js`?